include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -783,20 +809,32 @@ struct MapTypeInfo : detail::MapTypeInfoImpl<OrtMapTypeInfo> {
   ConstMapTypeInfo GetConst() const { return ConstMapTypeInfo{this->p_}; }
 };
 
-/// <summary>
-/// Type information that may contain either TensorTypeAndShapeInfo or
-/// the information about contained sequence or map depending on the ONNXType.
-/// </summary>
-struct TypeInfo : detail::Base<OrtTypeInfo> {
-  explicit TypeInfo(std::nullptr_t) {}                         ///< Create an empty TypeInfo object, must be assigned a valid one to be used
-  explicit TypeInfo(OrtTypeInfo* p) : Base<OrtTypeInfo>{p} {}  ///< C API Interop
+namespace detail {
+template <typename T>
+struct TypeInfoImpl : detail::Base<T> {
+  using B = Base<T>;
+  using B::B;
 
   ConstTensorTypeAndShapeInfo GetTensorTypeAndShapeInfo() const;  ///< Wraps OrtApi::CastTypeInfoToTensorInfo
   ConstSequenceTypeInfo GetSequenceTypeInfo() const;              ///< Wraps OrtApi::CastTypeInfoToSequenceTypeInfo
   ConstMapTypeInfo GetMapTypeInfo() const;                        ///< Wraps OrtApi::CastTypeInfoToMapTypeInfo
 
   ONNXType GetONNXType() const;
 };
+}  // namespace detail
+
+using ConstTypeInfo = detail::TypeInfoImpl<detail::Unowned<const OrtTypeInfo>>;
```

Comment:
>ConstTypeInfo = detail::Typ [](http://example.com/codeflow?start=6&length=27)

I initially omitted it. Is there a use for it?

Reply:
We need either `ConstTypeInfo` or `UnownedTypeInfo` to allow copying and passing of underlying `OrtTypeInfo`. I use it in the example found in onnxruntime/test/testdata/custom_op_openvino_wrapper_library

Reply:
Ok.